Assets/_BForBoss/_Utility/Editor/PropertyDrawers/ResolverPropertyDrawer.cs

```diff
@@ -0,0 +1,369 @@
+        private bool _includeInactiveGameObjects;
+        
+        private bool _isList = false;
+        private bool _isArray = false;
+        private bool _isDeclaredInStruct = false;
+        
+        //Button Content
+        private GUIContent _childResolveContent;
+        // private GUIContent _sceneResolveContent;
+        private GUIContent _parentResolveContent;
+        // private GUIContent _siblingResolveContent;
+        private GUIContent _selfResolveContent;
+        private GUIContent _clearContent;
+        
+        //Button Rects
+        private Rect _childResolveRect;
+        // private Rect _sceneResolveRect;
+        private Rect _parentResolveRect;
+        // private Rect _siblingResolveRect;
+        private Rect _selfResolveRect;
+        private Rect _clearContentRect;
+
+        private GUIStyle _clearContentGUIStyle;
+        
+        private enum ResolveType
+        {
+            FromChildren,
+            FromParent,
+            FromSelf
+        }
+        
+        
+        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        {
+            if (!IsResolvable(property))
+            {
+                EditorGUI.PropertyField(position, property, label);
+                return;
+            }
+
+            if (!_isContentInitialized)
+            {
+                InitializeContent();
+
+                ResolveAttribute resolveAttributeAttribute = (ResolveAttribute) attribute;
+                _includeInactiveGameObjects = resolveAttributeAttribute.IncludeInactive;
+            }
+            
+            //The way to tell if property is an array/list or not via property.propertypath
+            //https://answers.unity.com/questions/603882/serializedproperty-isnt-being-detected-as-an-array.html
+
+            Type fieldType = fieldInfo.FieldType;
+            _isArray = fieldType.IsArray;
+            _isList = fieldType.IsGenericType && fieldType.GetGenericTypeDefinition() == typeof(List<>);
+            
+            //Check if field is being declared within Struct
+            Type declaringType = fieldInfo.DeclaringType;
+            //https://stackoverflow.com/questions/1827425/how-to-check-programmatically-if-a-type-is-a-struct-or-a-class
+            _isDeclaredInStruct = declaringType != null && declaringType.IsValueType && !declaringType.IsPrimitive &&
+                                  declaringType != typeof(decimal) && declaringType != typeof(DateTime) &&
+                                  !declaringType.IsEnum;
+
+            position.Set(position.x, position.y, position.width - RESOLVER_WIDTH, position.height);
+            EditorGUI.PropertyField(position,property,label);
+
+            CreateRects(position);
+
+            DrawResolverButtons(property);
+        }
+
+        private void DrawResolverButtons(SerializedProperty property)
+        {
+            // Self (G) | Children (C) | Parent (P) | (OPTIONAL) Sibilings (R) | None (X)
```

Comment:
Can you update this comment. 
Comments are dangerous cuz if doesn't change. 👻

Reply:
> Disable it if it's a scriptableObject?
> 
> we can leave it to iteration 2
> 
> <img alt="Screen Shot 2021-12-31 at 12 57 55 PM" width="1229" src="https://user-images.githubusercontent.com/5779541/147834795-6ef3dca2-3b2d-49b2-b6ad-1bb5b0a1f378.png">

Hmm good point, will look into it in a couple of hours, and will let you know if we should leave it to iteration 2. Could be the same case as structs, where the fields themselves can have the attribute but the actual ScriptableObject declared field cannot

Reply:
Actually, @m4rtint  How is it supposed to work for ScriptableObjects? Since they are not Components or Gameobjects they cant just exist as a child, parent or self if that makes sense, They will always be a field of a component, instead of the actual component

Reply:
Nothing should happen when a [Resolve] is on the scriptableobjects